### DISCUSSION TITLE
1BRC with Oracle Database

### DISCUSSION THREAD

**Discussion by gvenzl** [gunnarmorling/1brc#707] [Show and tell]
I had (and still having) some fun with the 1BRC and Oracle Database.

Best results so far on external CSV file:

```sql
SELECT '{' ||
          LISTAGG(station_name || '=' || min_measurement || '/' || mean_measurement || '/' || max_measurement, ', ')
            WITHIN GROUP (ORDER BY station_name) ||
       '}' AS "1BRC"
 FROM
  (SELECT station_name,
          MIN(measurement) AS min_measurement,
          ROUND(AVG(measurement), 1) AS mean_measurement,
          MAX(measurement) AS max_measurement
    FROM measurements
     GROUP BY station_name
  );
```

* External file: 4.05s
* Internal table: 1.95s

Read more here: [1 billion row challenge in SQL and Oracle Database](https://geraldonit.com/2024/01/31/1-billion-row-challenge-in-sql-and-oracle-database/)

**Comment by alex-can-do-everything** [comment]
it was necessary to use 8 cores only
